src/main/kotlin/uk/gov/gdx/datashare/service/LegacyAdaptorInbound.kt

```diff
@@ -41,71 +42,80 @@ class LegacyAdaptorInbound(
     try {
       runBlocking {
         LockAssert.assertLocked()
-        S3Client {
-          region = "eu-west-2"
-        }.use { s3 ->
-          log.debug("Polling S3 bucket: $ingressBucket")
-          val listObjectsRequest = ListObjectsRequest {
-            bucket = ingressBucket
-          }
-          val objects = s3.listObjects(listObjectsRequest).contents
-
-          objects?.forEach { s3Object ->
+        log.debug("Polling S3 bucket: ${s3Config.ingressBucket}")
+        amazonS3.listObjectsV2(s3Config.ingressBucket).objectSummaries
+          .forEach { s3Object ->
             s3Object.key?.let {
-              processBucketObject(s3, it)
+              processBucketObject(it)
             }
           }
-        }
       }
-    } catch (e: NoSuchMethodError) {
-      log.error("Failed to connect to S3")
-    } catch (e: Exception) {
+    } catch (e: AmazonClientException) {
```

Comment:
I'm not totally sure what the try/catch here gets us if we're just logging it out - I think an uncaught exception in a scheduled task will get logged out fairly usefully for us anyway